compiler/luci-interpreter/include/luci_interpreter/MemoryManager.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LUCI_INTERPRETER_MEMORY_MANAGER_H
+#define LUCI_INTERPRETER_MEMORY_MANAGER_H
+
+#include "luci_interpreter/core/DataType.h"
+#include "luci_interpreter/core/Tensor.h"
+
+namespace luci_interpreter
+{
+
+class MManager
+{
+public:
+  virtual void allocate_memory(luci_interpreter::Tensor *tensor) = 0;
+  virtual void release_memory(luci_interpreter::Tensor *tensor) = 0;
```

Comment:
why not to use move semantics here?
e.g.
```suggestion
  virtual luci_interpreter::Tensor allocate_memory(luci_interpreter::Tensor &&tensor) = 0;
  virtual void release_memory(luci_interpreter::Tensor &&tensor) = 0;
```